osutil/exitcode.go

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) 2014-2020 Canonical Ltd
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License version 3 as
+// published by the Free Software Foundation.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package osutil
+
+import (
+	"os/exec"
+	"syscall"
+)
+
+// ExitCode extract the exit code from the error of a failed cmd.Run() or the
+// original error if its not a exec.ExitError
+func ExitCode(runErr error) (e int, err error) {
+	// golang, you are kidding me, right?
+	if exitErr, ok := runErr.(*exec.ExitError); ok {
+		waitStatus := exitErr.Sys().(syscall.WaitStatus)
+		e = waitStatus.ExitStatus()
+		return e, nil
```

Comment:
We could totally do this in a follow-up PR, but you no longer need this extra level of shenanigans and `syscall.WaitStatus`. I got the Go team to add [ProcessState.ExitCode](https://github.com/golang/go/issues/26539) in Go 1.12, so you can now do:


```suggestion
		e = exitErr.ProcessState.ExitCode()
		return e, nil
```